packages/adapter-xata/src/index.ts

```diff
@@ -0,0 +1,162 @@
+import type { Adapter } from "next-auth/adapters"
+
+import type { XataClient } from "./xata"
+
+export function XataAdapter(client: XataClient): Adapter {
+  return {
+    async createUser(user) {
+      const newUser = await client.db.nextauth_users.create(user)
+      return newUser
+    },
+    async getUser(id) {
+      const user = await client.db.nextauth_users.filter({ id }).getFirst()
+      return user ?? null
+    },
+    async getUserByEmail(email) {
+      const user = await client.db.nextauth_users.filter({ email }).getFirst()
+      return user ?? null
+    },
+    async getUserByAccount({ providerAccountId, provider }) {
+      const result = await client.db.nextauth_users_accounts
+        .select(["user.*"])
+        .filter({
+          "account.providerAccountId": providerAccountId,
+          "account.provider": provider,
+        })
+        .getFirst()
+      const user = result?.user
+      return user ?? null
+    },
+    async updateUser(user) {
+      const result = await client.db.nextauth_users.update(user.id!, user)
+      return (
+        result ?? {
+          ...user,
+          id: user.id!,
+          emailVerified: user.emailVerified ?? null,
+        }
+      )
+    },
+    async deleteUser(id) {
+      return await client.db.nextauth_users.delete(id)
+    },
+    async linkAccount(initialAccount) {
+      const { userId, ...account } = initialAccount
+      const newXataAccount = await client.db.nextauth_accounts.create({
+        ...account,
+        user: { id: userId },
+      })
+      await client.db.nextauth_users_accounts.create({
+        user: { id: userId },
+        account: { id: newXataAccount.id },
+      })
+    },
+    async unlinkAccount({ providerAccountId, provider }) {
+      /**
+       * @todo refactor this when we support DELETE WHERE.
+       */
+      const connectedAccount = await client.db.nextauth_users_accounts
+        .filter({
+          "account.providerAccountId": providerAccountId,
+          "account.provider": provider,
+        })
+        .getFirst()
+
+      if (!connectedAccount) {
+        return
+      }
+      return await client.db.nextauth_users_accounts.delete(connectedAccount.id)
+    },
+    async createSession(initialSession) {
+      const { userId, ...session } = initialSession
+      const newXataSession = await client.db.nextauth_sessions.create({
+        ...session,
+        user: { id: userId },
+      })
+      await client.db.nextauth_users_sessions.create({
+        user: { id: userId },
+        session: { id: newXataSession.id },
+      })
+      return { ...newXataSession, ...session, userId }
```

Comment:
Shouldn't we spread the `newXataSession` after the initial session? 🤔

Reply:
Yes, good catch!

Reply:
```suggestion
      return { ...session, ...newXataSession, userId }
```